tests/unit/development/test_base_component.py

```diff
@@ -508,3 +508,29 @@ def update(v1, v2):
     assert input2.id == "23a7711a-8133-2876-37eb-dcd9e87a1613"
     # we make sure that the if the id is set explicitly, then it is not replaced by random id
     assert output3.id == "output-3"
+
+
+def test_debc029_random_id_errors():
+    app = Dash(__name__)
+
+    input1 = dcc.Input(value="Hello Input 1", persistence=True)
```

Comment:
I am pretty sure its probably late, and not sure if its valid here. But we can force the user to always Instantiate `Input` with a defined id by making it a required keyword-arg. That way it will simply be a compile error. 

```
class Input():
        def __init__(self, value=None, style=None, *, id):
                ...
                
        
# The following line will throw an error 
# The error states that __init__ is missing a required keyword-arg 'id'

input1 = Input(value="something", style={'color': 'orange'})

              
```

Reply:
cc @alexcjohnson @chriddyp

Reply:
Basically anything after the `*` becomes a required keyword arg. and its part of python 3.6.

Reply:
Dash components do have the ability to specify required args - and this gets handled in each component's `__init__` via the component generator. The challenge here is that it's coupled, though you're right that in principle we could handle that in `__init__`, but it would have to be explicit code. This would prevent creating the component and then later setting its ID, but not sure if that's really a pattern we want to promote.